zookeeper/core/component_test.py

```diff
@@ -409,3 +401,38 @@ class A3:
         captured.err
         == "WARNING: Unable to check that typing.Tuple[int, int, int] is a sub-type of typing.Tuple[float, float, float].\n"
     )
+
+
+def test_component_configure_error_non_existant_key():
+    @component
+    class Child:
+        a: int = Field(1)
+
+    @component
+    class Parent:
+        b: str = Field("foo")
+        child: Child = ComponentField(Child)
+
+    @component
+    class GrandParent:
+        c: float = Field(3.14)
+        parent: Parent = ComponentField(Parent)
+
+    # A missing key should cause an error.
+    with pytest.raises(
+        ValueError,
+        match="Key 'd' does not correspond to any field of component 'GrandParent'.",
+    ):
+        configure(GrandParent(), {"d": "bar"}, name="GrandParent")
+
+    # 'b' exists on the parent but not the grand-parent, so should be set via
+    # `parent.b='bar'`. This should raise an error:
+    with pytest.raises(
+        ValueError,
+        match="Key 'b' does not correspond to any field of component 'GrandParent'.",
+    ):
+        configure(GrandParent(), {"b": "bar"})
+    # But this should not:
+    g = GrandParent()
+    configure(g, {"parent.b": "bar"})
+    assert g.parent.b == "bar"
```

Comment:
maybe add a test to ensure that {"parent.non_existent_field": "bar"} does fail?

Reply:
I've added that test in 78cbcda.